src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Mapping/RazorMapToDocumentEditsEndpointTest.cs

```diff
@@ -0,0 +1,559 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.LanguageServer.Mapping;
+using Microsoft.AspNetCore.Razor.Telemetry;
+using Microsoft.AspNetCore.Razor.Test.Common;
+using Microsoft.AspNetCore.Razor.Test.Common.LanguageServer;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Razor.DocumentMapping;
+using Microsoft.CodeAnalysis.Razor.Protocol;
+using Microsoft.CodeAnalysis.Razor.Protocol.DocumentMapping;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Test.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Test.Mapping;
+
+public class RazorMapToDocumentEditsEndpointTest : LanguageServerTestBase
+{
+    private readonly IDocumentMappingService _documentMappingService;
+
+    public RazorMapToDocumentEditsEndpointTest(ITestOutputHelper testOutputHelper) : base(testOutputHelper)
+    {
+        _documentMappingService = new LspDocumentMappingService(
+            FilePathService,
+            new TestDocumentContextFactory(),
+            LoggerFactory);
+    }
+
+    [Fact]
+    public Task SimpleEdits_Apply()
+        => TestAsync(
+        csharpSource: """
+        class MyComponent : ComponentBase
```

Comment:
Nit: Indenting the actual string contents one level in from the parameter name makes it much easier to parse.

Applies the whole file

Reply:
Done